CMakeLists.txt

```diff
@@ -1,4 +1,8 @@
-cmake_minimum_required(VERSION 3.9)
+if(APPLE)
+  cmake_minimum_required(VERSION 3.12)
```

Comment:
Is there a reason why we need to have a later version for all Apple devices?
Or is it only for AppleClang?

Reply:
I needed it for linking to targets for AppleClang. I dont need it anymore, so I have removed it.